<launch>
    <node name="marker_cam" pkg="usb_cam" type="usb_cam_node" output="screen" respawn="true" respawn_delay="1">
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="marker_cam" />
        <param name="camera_name" value="leopard_ov7725_fov90" />
        <param name="camera_info_url" value="file://$(find shining_ros)/param/leopard_ov7725_fov90.yaml" />
        <param name="framerate" value="10.0" />
        <param name="io_method" value="mmap"/>
    </node>

        <arg name="parameters_path" default="$(find shining_ros)/param/ice"/>

        <arg name="markerSize"      default="0.114"/>    <!-- in m -->
        <arg name="camera_name"     default="marker_cam"/>

        <!-- Parameters for detecting erroneous conditions -->
        <arg name="min_distance"      default="0.5"/>   <!-- in m, min distance of detection -->
        <arg name="max_distance"      default="1.25"/>    <!-- in m, max distance of detection -->
        <arg name="yaw_tolerance"     default="0.4"/>   <!-- in radians, for detecting twisted code / up-side down -->
        <arg name="roll_tolerance"    default="0.4"/>    <!-- in radians, approaching angle (sideway) -->
        <arg name="pitch_tolerance"   default="0.5"/>    <!-- in radians, to make sure code is on a flat surface -->
        <arg name="overlay_error_message"  default="true"/>  <!-- boolean, overlay error on camera image -->

        <node name="image_proc" pkg="image_proc" type="image_proc" ns="$(arg camera_name)" required="true"/>

        <node pkg="aruco_ros" type="aruco_detector" name="multi_aruco_detector" required="true">
            <rosparam file="$(arg parameters_path)/aruco_codes.yaml" command="load"/>
            <remap from="/image" to="/$(arg camera_name)/image_rect" />
            <remap from="/camera_info" to="/$(arg camera_name)/camera_info" />
            <param name="image_is_rectified" value="True"/>
            <param name="marker_size"        value="$(arg markerSize)"/>
            <param name="reference_frame"    value="$(arg camera_name)"/>
            <param name="camera_frame"       value="$(arg camera_name)"/>
            <param name="marker_frame"       value="aruco_marker_frame" />
            <param name="min_distance"       value="$(arg min_distance)" />
            <param name="max_distance"       value="$(arg max_distance)" />
            <param name="yaw_tolerance"      value="$(arg yaw_tolerance)" />
            <param name="roll_tolerance"     value="$(arg roll_tolerance)" />
            <param name="pitch_tolerance"    value="$(arg pitch_tolerance)" />
            <param name="overlay_error_message"       value="$(arg overlay_error_message)" />

        </node>

</launch>
